msupply-foundation/open-msupply
# 3191 generate other returns half

Fixes #3191

# 👩🏻‍💻 What does this PR do? 
 

Expands the inbound/outbound shipment transfer processors to also support returns.

Updates some of the inbound return detail UI now that linkedShipment is available (to be updated to linkedInvoice or something better named!)

# 🧪 How has/should this change been tested? 


- Create an outbound return, and set it to picked
- Switch to the store you created the return for - you should see a generated inbound return
- Switch to the outbound return, set to shipped
- Switch to inbound return - can now edit
- Deleting a picked outbound return should also delete the generated inbound
- Outbound UI doesn't have support for comments/references yet but can test this if you edit DB directly 😁 
- Create a supplier credit in OG, see that that the resulting Inbound Return displayed in omSupply


## 💌 Any notes for the reviewer?

Things I need to ask about tomorrow:
- [x] How do I test this syncs between different sites?
  - [x] Return reasons are configured centrally, so the return_reason_id will map to the same reason for everyone right??
- [x] How do I test this maps correctly from OG to OMS?
- [x] I think OMS to OG looks all good, but it's also possible I'm missing something, would be good to validate with someone
- [x] Do we want the linked requisition to get passed through to the return as well? In that case, I think we'll need to store the requisition_id when we create the returns?
